LibrarySystem/Queries/12.TriggerImplementation.sql

```diff
@@ -0,0 +1,31 @@
+CREATE TABLE AuditLog 
+(	
+	LogID INT IDENTITY(1,1) PRIMARY KEY,
+    BookID INT,
+    StatusChange VARCHAR(255),
+    ChangeDate DATETIME
+);
+
+GO
+CREATE TRIGGER tr_CurrentStatusChange
+ON Books
+AFTER UPDATE
+AS
+BEGIN
+    IF UPDATE(CurrentStatus)
+    BEGIN
+        INSERT INTO AuditLog (BookID, StatusChange, ChangeDate)
+        SELECT
+            i.BookID,
+            CASE
+                WHEN i.CurrentStatus = 'Available' AND d.CurrentStatus = 'Borrowed' THEN 'Borrowed to Available'
```

Comment:
No need to track the status change that way, 'Borrowed to Available', just add the UpdatedStatus, ;ex: 'Available.